### ISSUE TITLE
[HOLD Issue #309277][$1000] LHN - Chat that doesn't have a message does not disappears from the LHN

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#22806]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

**Scenario "empty DMs"**
1. Go to https://staging.new.expensify.com/
2. Login with any account
3. Search for a user that you don't have any messages with and open the chat. Don't send any message
4. Navigate away from the conversation by clicking any other available conversation in the LHN list

**Scenario "empty threads"**
1. Go to https://staging.new.expensify.com/
2. Login with any account
3. Select any user
4. Select "Reply in Tread" option

## Expected Result:
1. Conversation with no messages should disappear from LHN conversation list
2. Thread report is not displayed until the comment is posted

## Actual Result:
1. Conversation that doesn't have any messages doesn't disappear from from LHN
2. Thread report is displayed when the comment is not posted

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.37.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

**Scenario "empty DMs"**

https://github.com/Expensify/App/assets/78819774/dacf6489-8f37-4f7f-99b7-dbadc42cb41f

**Scenario "empty threads"**

https://github.com/Expensify/App/assets/93399543/74a0a9f2-16d7-4194-a7bc-320ea8633dc7

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013fc034f8ebadf86c</li>
        <li>Upwork Job ID: 1681415083049984000</li>
        <li>Last Price Increase: 2023-08-01</li>
    </ul>
</details>

**Comment by alexpensify** [comment]
I'll review soon and update.

**Comment by alexpensify** [comment]
This one is on my testing list for the week.

**Comment by kbecciv** [comment]
Hi @alexpensify We have Slack discussion about the opening "empty threads" and for starting "empty DMs" https://expensify.slack.com/archives/C049HHMV9SM/p1689466696046149, how would you like me to handle the opening "empty threads? Would you like me to create another GH ticket or combine? David Barrett would like to fully test that shared solution on both flows. Thank you in advance.

**Comment by alexpensify** [comment]
@kbecciv - let's add the additional flow to this GH. Please update the original GH to include the flow that David mentioned,  Thanks!

**Comment by alexpensify** [comment]
Adding the `external` label for now. @rushatgabhane - could you please review the proposal and identify if it will work to address both flows? Thanks!

**Comment by kbecciv** [comment]
@alexpensify I added two scenarios, please let me know if anything else I need to do. Thanks

**Comment by alexpensify** [comment]
@quinthar - you were part of the Slack thread that mentioned the other flow. Can you share more feedback for @dukenv0307's questions? Thanks!

**Comment by alexpensify** [comment]
I'll start a discussion tomorrow to confirm the flows.

**Comment by alexpensify** [comment]
I started a discussion here: https://expensify.slack.com/archives/C049HHMV9SM/p1690236131313079?thread_ts=1689466696.046149&cid=C049HHMV9SM

**Comment by quinthar** [comment]
Hi there, great questions.  First, I'd like to expand the problem description.  The problem consists of two parts:

Problem 1
----
When you create a DM/thread, it is shared with the other party immediately, before you even type.  It appears as unread on their LHN, which reveals that you are _thinking_ about writing something to them, but haven't yet.  This is a confusing privacy violation.

Problem 2
----
When you a DM/thread, if you choose not to post anything, you and the party still see it in your LHNs forever.

Solution
----
Do not notify the server of the new DM/thread until after the user has actually posted the first message.  If they do not post a message, then the server should never know that you even considered it.

**Comment by alexpensify** [comment]
@dukenv0307 are there other questions or are you good to prepare your updated proposal that includes both flows?

**Comment by rushatgabhane** [comment]
> If the user creates a new room without a message, could we hide the room report in LHN?

i think we shouldn't hide room even if there is no message

**Comment by alexpensify** [comment]
Not overdue, the proposal will go through another round of reviews.

**Comment by alexpensify** [comment]
Update: waiting for the proposal review here.

**Comment by alexpensify** [comment]
@rushatgabhane - any update here? If no reply by tomorrow, I will be looking to reassign. Thank you!

**Comment by rushatgabhane** [comment]
@dukenv0307 did you forget to update your proposals?

**Comment by rushatgabhane** [comment]
> . User A still notifies to server but the server doesn't notify user B

@dukenv0307 will that require backend changes?

**Comment by cristipaval** [comment]
Alright

> @rushatgabhane @quinthar In case User A opens a new chat with User B, in order not to refactor too much code, I suggest that:
> 
> 1. The main screen of user A will redirect to a new chat with user B and the new report will appear in LHN. User A still notifies to server but the server doesn't notify user B.
> 2. If User A click to another chat, the main screen will redirect and the report with user B will disappear in LHN
> 3. If User A send a message to user B, user A will notify to Server and Server will notify to user B.

@dukenv0307 I totally get your challenge in solving the problem in the front-end only. Thanks for your suggested solution. 
I still wouldn't create a new report in the backend before the user posts anything in the chat. @Beamanator has more knowledge about how user details work and he has a great idea. What if the front end fetches the avatar and display name from the backend when a new chat is opened and there's nothing posted in the chat yet? And we stick with the plan to notify the server about the new chat only when the user posts something in it. Does this solution work?

**Comment by rushatgabhane** [comment]
yeah precisely. i agree! 

> Maybe we can have an endpoint to get the correct avatar, display name, and accountID. It shouldn't write to the database.

i suggested the same thing here https://github.com/Expensify/App/issues/22806#issuecomment-1661839008

**Comment by cristipaval** [comment]
> yeah precisely. i agree!
> 
> > Maybe we can have an endpoint to get the correct avatar, display name, and accountID. It shouldn't write to the database.
> 
> i suggested the same thing here [#22806 (comment)](https://github.com/Expensify/App/issues/22806#issuecomment-1661839008)

aaaah, I missed your comment there. Good one 👍

**Comment by cristipaval** [comment]
> @rushatgabhane @cristipaval From the last discussion
> 
> > I still wouldn't create a new report in the backend before the user posts anything in the chat
> 
> I see that we can't get the correct avatar, display name, and accountID. We can only create the default report from FE and after the user sends the first message, we will call API from the server and update the correct avatar, display name, and accountID. Please correct me if I am wrong

The idea is to create a new command that front-end hits when _opens_ a new chat in front-end:
1. Bob never texted Alice again
2. Bob searches for `alice@email.com`
3. Front-end creates an optimistic chat and fetches the avatar and display name for `alice@email.com` to show them in the optimistically created chat
4. If Bob leaves the chat, it disappears from LHN
5. If Bob posts anything in the chat, the front-end notifies about the chat, and the message

Does this make sense @dukenv0307 @rushatgabhane ?

**Comment by alexpensify** [comment]
Thank you @cristipaval for this context!

**Comment by rushatgabhane** [comment]
sounds good! @dukenv0307 @cristipaval maybe we can work on the frontend in parallel ?

**Comment by cristipaval** [comment]
> Make sense to me. With this approach, I have a bit curios:
> 
> 1. we need to create a new API to get an avatar and display name for [alice@email.com](mailto:alice@email.com) (step 3).

That's correct.

> 2. 2. In step 5, we need to call 2 API OpenReport and AddComment.
>       @rushatgabhane @cristipaval  Please correct me If I am wrong

You're correct, but this violates our 1:1:1 pattern, we want to have only 1 API call for a user action. We'll update the `OpenReport` API command to pass the first message to that report.

**Comment by alexpensify** [comment]
Not overdue - @rushatgabhane when you get a chance can you please review the most recent proposal? Thanks!

**Comment by rushatgabhane** [comment]
> and avatar based on email

to be pedantic - based on `login`* (it could be email or phone)

**Comment by rushatgabhane** [comment]
> We should hide 1:1 chat if there is no message

no. we hide the chat only if you leave it without any message

**Comment by alexpensify** [comment]
@rushatgabhane when you get a chance, can you please review the updated proposal? Thanks!

**Comment by rushatgabhane** [comment]
@dukenv0307 thank you, your proposal looks crisp!
🎀 👀 🎀 C+ reviewed https://github.com/Expensify/App/issues/22806#issuecomment-1670888065

**Comment by cristipaval** [comment]
The first API change is currently in review.

**Comment by cristipaval** [comment]
It seems that this issue is going to be fixed by @srikarparsi's [project](https://expensify.slack.com/archives/CC7NECV4L/p1691616201758119). @srikarparsi Could you please confirm? Do you have an issue on which I could hold this one?

**Comment by srikarparsi** [comment]
Hey @cristipaval! Yes, this should be fixed by the [Improving Threads Logic](https://github.com/Expensify/Expensify/issues/309277) project. You can use that issue to hold this one :)

**Comment by alexpensify** [comment]
Thanks, since this is on hold, I'm moving to Weekly.

**Comment by alexpensify** [comment]
Weekly Update: On Hold

**Comment by cristipaval** [comment]
@alexpensify could you confirm and close the issue?

**Comment by alexpensify** [comment]
I'm catching up from being OOO, I will test tomorrow.

**Comment by alexpensify** [comment]
Still on my testing list

**Comment by alexpensify** [comment]
Yeah, I confirmed too, and will close this GH.

**Comment by alexpensify** [comment]
@dukenv0307 - I'll review the feedback and follow up later this week.

**Comment by alexpensify** [comment]
Still on my review list

**Comment by alexpensify** [comment]
Thanks for your patience here. Since the examples shared [here](https://github.com/Expensify/App/issues/22806#issuecomment-1708629741), there have been a few updates to the partial payment process. I need to ask a few questions and need your opinions: 

1. What compensation do you think is the percent due here for the work input here - 0, 25, 50 and 100%
2. Reasoning why has been shared [here](https://github.com/Expensify/App/issues/22806#issuecomment-1708629741). 

@dukenv0307 and @rushatgabhane - please reply, Thanks!

**Comment by alexpensify** [comment]
Thanks for sharing @dukenv0307! 

@rushatgabhane - can you share some feedback or if you agree? Thanks!

**Comment by Christinadobrzyn** [comment]
Also reached out to BZ about payment - https://expensify.slack.com/archives/C01SKUP7QR0/p1695743819720659

**Comment by alexpensify** [comment]
@rushatgabhane I should have included this before but there is a process for partial payments. The summary is: 

>Ask the C+ or contributor what compensation they think they’re due and ask them to provide reasoning why. Also ask them to present their compensation request as a percent of the job price where standard payments are 0, 25, 50 and 100%.

From there, I will review and plan accordingly. I'd still appreciate if you shared a percent here. Thanks!

**Comment by alexpensify** [comment]
Not overdue, waiting for feedback. @rushatgabhane when you get a chance can you share the percentage number? Thanks!

**Comment by cristipaval** [comment]
Just sharing my thoughts here:

- The C+ did review the proposals and jumped into the conversation, but there was no real code review or testing. 25% should be fine for them.
- The assigned contributor spent some time on the accepted solution and encountered some challenges. They raised them here, and there was some back and forth. Paying 100% is fair in this case.

**Comment by alexpensify** [comment]
Thanks, I will start this payment breakdown tomorrow.

**Comment by alexpensify** [comment]
Here is the payment summary: 

-  External issue reporter N/A 
-  Contributor that fixed the issue @dukenv0307 $1000
-  Contributor+ that helped on the issue and/or PR @rushatgabhane $250

Upwork Job: https://www.upwork.com/jobs/~01b0decd73ba4f11c0

*If applicable, the bonuses will be applied on the final payment 

Extra Notes regarding payment: This payment breakdown has been in discussion but here is the summary https://github.com/Expensify/App/issues/22806#issuecomment-1756471196

**Comment by alexpensify** [comment]
@dukenv0307 - I've sent a proposal via Upwork. Please accept and I can complete the process, thanks!

**Comment by JmillsExpensify** [comment]
$250 payment approved for @rushatgabhane based on BZ summary.

**Comment by alexpensify** [comment]
Everyone has been paid! I've completed the process in Upwork.
